core/state_transition.go

```diff
@@ -304,6 +307,28 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 	}
 	// End Quorum - Privacy Enhancements
 
+	// do the affected contract managed party checks
+	if msg, ok := msg.(PrivateMessage); ok && isQuorum && st.evm.SupportsMultitenancy && msg.IsPrivate() {
```

Comment:
`isQuorum` in the condition may already clarify the intention already without the need of explicit comments. 

Also for a fairly complex logic meant for Quorum, I would rather add some documentation at the method level than inline as it would provide a better overview of what've been changed.